lib/net.js

```diff
@@ -1163,11 +1163,11 @@ var createServerHandle = exports._createServerHandle =
     debug('bind to ' + (address || 'anycast'));
     if (!address) {
       // Try binding to ipv6 first
-      err = handle.bind6('::', port);
+      err = handle.bind6('::1', port);
       if (err) {
         handle.close();
         // Fallback to ipv4
-        return createServerHandle('0.0.0.0', port);
+        return createServerHandle('localhost', port);
```

Comment:
Though it does require a dns check. Not much extra I realize, but just placing the reminder.

@bnoordhuis It seems to me using localhost is a regression. If no host is provided but type 4 is passed  then it's possible to still get a 6 address back. And binding to ::1 does not allow connections from 127.0.0.1.